### PR TITLE
VEOSS-1823 Dashboard Variation for Excluded Traders That Have Left Service for 3 Years

### PR DESCRIPTION
**Pull request by ZDevelop94**

### CODE REVIEW

File: app/controllers/YourAccountController.scala [hunk 1]
```diff
@@ -66,15 +71,19 @@ class YourAccountController @Inject()(
       }
   }
 
-  private def normalView()(implicit request: RegistrationRequest[AnyContent]): Future[Result] = {
+  private def normalView()(implicit request: RegistrationRequest[AnyContent], clock: Clock): Future[Result] = {
```
Comment (hmrcvmu) [inline app/controllers/YourAccountController.scala:74]: `clock` can be moved at the class level

File: app/controllers/YourAccountController.scala [hunk 2]
```diff
@@ -120,64 +129,77 @@ class YourAccountController @Inject()(
     }
   }
 
-  private def prepareViewWithFinancialData(returnsViewModel: Seq[Return],
+  private def prepareViewWithFinancialData(nonExcludedReturns: Seq[Return],
+                                           excludedReturns: Seq[Return],
                                            currentPayments: CurrentPayments,
-                                           periodInProgress: Option[Period])(implicit request: RegistrationRequest[AnyContent]): Future[Result] = {
+                                           periodInProgress: Option[Period])(implicit request: RegistrationRequest[AnyContent], clock: Clock): Future[Result] = {
```
Comment (hmrcvmu) [inline app/controllers/YourAccountController.scala:135]: `clock` can be moved to the class level

File: app/controllers/YourAccountController.scala [hunk 3]
```diff
@@ -120,64 +129,77 @@ class YourAccountController @Inject()(
...
       ))
     }
   }
 
-  private def prepareViewWithNoFinancialData(returnsViewModel: Seq[Return], periodInProgress: Option[Period])
-                                            (implicit request: RegistrationRequest[AnyContent]): Future[Result] = {
+  private def prepareViewWithNoFinancialData(returnsViewModel: Seq[Return],
+                                             excludedReturns: Seq[Return],
+                                             periodInProgress: Option[Period])
+                                            (implicit request: RegistrationRequest[AnyContent], clock: Clock): Future[Result] = {
```
Comment (hmrcvmu) [inline app/controllers/YourAccountController.scala:174]: `clock` can be moved to the class level

File: app/controllers/YourAccountController.scala [hunk 4]
```diff
@@ -55,6 +58,8 @@ class YourAccountController @Inject()(
   def onPageLoad: Action[AnyContent] = cc.authAndGetRegistration.async {
     implicit request =>
 
+      implicit val c: Clock = clock
```
[the remark this answers is not in the document]
Comment (ZDevelop94) [reply app/controllers/YourAccountController.scala:61]: it is but its not implicit

File: app/controllers/YourAccountController.scala (hunk 1 above)
Comment (ZDevelop94) [reply app/controllers/YourAccountController.scala:74]: it is but its not implicit

File: app/controllers/YourAccountController.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (ZDevelop94) [reply app/controllers/YourAccountController.scala:61]: I wanted to do this originally but I thought Play could only pass the clock explicitly according to the routes and would have to change a lot of implementations.

**Review by hmrcvmu — DISMISSED** [review]

File: app/controllers/YourAccountController.scala [hunk 5]
```diff
@@ -58,6 +59,8 @@ class YourAccountController @Inject()(
                                      )(implicit ec: ExecutionContext)
   extends FrontendBaseController with I18nSupport with Logging {
 
+  implicit val c: Clock = clock
```
Comment (hmrcvmu) [inline app/controllers/YourAccountController.scala:62]: The clock could be moved to the list of implicits: `(implicit ec: ExecutionContext, clock: Clock)`

**Review by hmrcvmu — APPROVED** [review]

**Pull request merged**